roles/openshift_health_checker/library/search_journalctl.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/python
+"""Interface to journalctl"""
+
+from time import time
+from datetime import datetime
+import json
+import re
+import subprocess
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+class InvalidMatcherRegexpException(Exception):
+    """Base exception class for invalid matcher regexp fields"""
+    def __init__(self, message):
+        Exception.__init__(self, message)
+
+
+def stamp_too_old(stamp_microseconds, stamp_limit_seconds):
+    """Receives a log message time stamp (in microseconds) and determines if it
+    is newer or older than stamp_limit_seconds"""
+    stamp_seconds = float(stamp_microseconds) / 1000000
+    stamp_actual = datetime.fromtimestamp(stamp_seconds)
+    stamp_limit = datetime.fromtimestamp(stamp_limit_seconds)
+    return stamp_actual < stamp_limit
+
+
+def find_matches(log_input, matcher, log_count_limit, timestamp_limit_seconds):
+    """Receives iterable input, a matcher dictionary and scans journalctl at the
+    given systemd module for log messages matching a given regular expression.
+    Returns the matched log message (or None) and an error string if any."""
+    try:
+        regexp = re.compile(matcher["regexp"])
+        start_regexp = re.compile(matcher["start_regexp"])
+    except re.error as err:
+        msg = "A log matcher object was provided with an invalid regular expression: {}"
+        raise InvalidMatcherRegexpException(msg.format(str(err)))
+
+    log_count = 0
+    matched = None
+
+    for line in log_input:
+        log_count += 1
+        if log_count >= log_count_limit:
+            break
+
+        try:
+            obj = json.loads(line.rstrip())
```

Comment:
Why `rstrip()`?

If that's because of a newline character at the end of the line, removing it is unnecessary:

```
In [4]: import json

In [5]: json.loads('{"line": 54}\n')
Out[5]: {u'line': 54}

In [6]: json.loads('{"line": 54}\n\n')
Out[6]: {u'line': 54}

In [7]: json.loads('{"line": 54}\n   \n')
Out[7]: {u'line': 54}
```